src/protocol/libp2p/kademlia/handle.rs

```diff
@@ -364,6 +370,15 @@ impl KademliaHandle {
         query_id
     }
 
+    /// Stop providing the key on the DHT.
+    ///
+    /// This will stop republishing the provider, but won't
+    /// remove it instantly from the nodes. It will be removed from them after the provider TTL
+    /// expires, set by default to 48 hours.
```

Comment:
dq: Do you think we should consider an eviction strategy from the memory store that removes older but not yet expired records similar to LRU?